client/utils/utils.go

```diff
@@ -119,7 +119,9 @@ func EnrichWithGas(txBldr authtxb.TxBuilder, cliCtx context.CLIContext, msgs []s
 
 // CalculateGas simulates the execution of a transaction and returns
 // both the estimate obtained by the query and the adjusted amount.
-func CalculateGas(queryFunc func(string, common.HexBytes) ([]byte, error), cdc *amino.Codec, txBytes []byte, adjustment float64) (estimate, adjusted uint64, err error) {
+func CalculateGas(queryFunc func(string, common.HexBytes) ([]byte, error),
```

Comment:
I'd like to see a written explanation of your thinking in a logical explanation of the aesthetics and functional purposes or arranging the code otherwise, I would gladly take on a debate of this form, I would like to see a uniform style within the SDK meaning we should come to an agreement over this. A sigh doesn't insight much out of me if it's not paired with something substantial.

Reply:
1. This is Go...we shouldn't be debating formatting and aesthetics. If we are, we're doing something wrong. As far as logic goes, let's just say it took me at least 5 seconds to comprehend this function signature.

From an excellent [blog](https://dave.cheney.net/practical-go/presentations/qcon-china.html) which you should read if you haven't:
```
My rule of thumb is; if it fits through gofmt then its usually 
not worth holding up a code review for.
```

2. I highly doubt we're ever going to have a uniform agreed upon style because we'll likely have very conflicting opinions, but mostly due to reason (1) above.

3. I wasn't sighhhing due to your changes actually, but due to the already existing complicated and unreadable nature of the function signature.

tl;dr ignore my sigh and we shouldn't hold up PRs due to style (we're never going to agree anyway).

Reply:
I disagree that we shouldn't be exploring code aesthetics... as it can lead to more readable code. as you've just pointed out it took you 5 seconds to comprehend this function signature - sounds dysfunctional to me! I think that if we both keep an open mind we can easily arrive at consensus, I've certainly got my mind open I'm not emotionally attached to the way I currently format things... However I also will not accept anyone's authority on the matter unless there is acceptable logical reasoning behind it - something which I've justified for myself (however there may be better contradictory justifications as well) 

checking out that blog post now.

Reply:
I think one of the most confusing things about this function header is the fact that there is a function variable defined _within_ the function header, and that this inner-defined-function's output variables line up with the output variables of the function header

Reply:
+1. I think any inline function arguments to functions should have a type alias like you just suggested. Much easier to read ^__^